modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalResponse.java

```diff
@@ -61,8 +61,22 @@ public RankEvalResponse(double metricScore, Map<String, EvalQueryQuality> partia
         this.failures = new HashMap<>(failures);
     }
 
-    RankEvalResponse() {
-        // only used in RankEvalAction#newResponse()
+    RankEvalResponse(StreamInput in) throws IOException {
+        super(in);
+        this.metricScore = in.readDouble();
```

Comment:
It would be nice to start marking these instances variables `final`. I think this PR used an automated script though, so that might be challenging -- maybe we can work on immutability in a future refactoring effort.

Reply:
Yes I think we can use eg intellij reports to find members that can be made final. There are a lot after all of these PRs.